### PR TITLE
Provision a new user with scim1.1  using unauthorized credentials

### PR DESCRIPTION
**Pull request by shavantha**
Purpose
This test case has the following scenario
1.The admin adds a user ex:(scim1user) using the scim1.1 API
2.We attempt to create a second user ex: (scim2user) using the credentials of scim1user
3.Since scim1user does not have adequate permission we assert for  [401] User is not authorized to perform provisioning expected.

Known Issues
https://github.com/wso2/product-is/issues/4022

Tested environments
1.Ubuntu18.04
2.Java: 1.8.0_161
3.Identity Server fresh pack with embedded ldap

### CODE REVIEW

File: product-scenarios/1-manage-users-roles/1.1-user-registration-to-an-application/1.1.1-user-registration-with-web-application-itself/1.1.1.1-provision-user-using-SCIM1.1/src/test/java/org/wso2/identity/scenarios/test/scim/UserProvisionWithInsufficientPrivilegesSCIMTestCase.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    public void testInit() throws Exception {
+
+        setKeyStoreProperties();
+        client = HttpClients.createDefault();
+        super.init();
+        scimUsersEndpoint = backendURL + SCIMConstants.SCIM_ENDPOINT + SEPERATOR + Constants.SCIM_ENDPOINT_USER;
+    }
+
+    @Test(description = "1.1.1.1.11")
+    public void testSCIMCreateFirstUser() throws Exception {
```
Comment (sashikamw) [inline product-scenarios/1-manage-users-roles/1.1-user-registration-to-an-application/1.1.1-user-registration-with-web-application-itself/1.1.1.1-provision-user-using-SCIM1.1/src/test/java/org/wso2/identity/scenarios/test/scim/UserProvisionWithInsufficientPrivilegesSCIMTestCase.java]: Use this method as a dependent for the main test method

**Comment by shavantha** [comment]
closing the PR until the cleanup

**Comment by shavantha** [comment]
Please keep this PR un merged until the below issue is fixed
https://github.com/wso2/product-is/issues/4022

File: product-scenarios/1-manage-users-roles/1.1-user-registration-to-an-application/1.1.1-user-registration-with-web-application-itself/1.1.1.1-provision-user-using-SCIM1.1/src/test/java/org/wso2/identity/scenarios/test/scim/UserProvisionWithInsufficientPrivilegesSCIMTestCase.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    private void testSCIMCreateFirstUser() throws Exception {
+
+        JSONObject userObject = new JSONObject();
+        JSONArray schemas = new JSONArray();
+        userObject.put(SCIMConstants.SCHEMAS_ATTRIBUTE, schemas);
+        JSONObject names = new JSONObject();
+        names.put(SCIMConstants.GIVEN_NAME_ATTRIBUTE, SCIMConstants.GIVEN_NAME_CLAIM_VALUE);
+        userObject.put(SCIMConstants.NAME_ATTRIBUTE, names);
+        userObject.put(SCIMConstants.USER_NAME_ATTRIBUTE, SCIMConstants.USERNAME);
+        userObject.put(SCIMConstants.PASSWORD_ATTRIBUTE, SCIMConstants.PASSWORD);
```
Comment (ayshsandu) [inline product-scenarios/1-manage-users-roles/1.1-user-registration-to-an-application/1.1.1-user-registration-with-web-application-itself/1.1.1.1-provision-user-using-SCIM1.1/src/test/java/org/wso2/identity/scenarios/test/scim/UserProvisionWithInsufficientPrivilegesSCIMTestCase.java]: Line64 to 71, are duplicate again from line 80 to 87. Can externalize to a private method.

Comment (shavantha) [reply product-scenarios/1-manage-users-roles/1.1-user-registration-to-an-application/1.1.1-user-registration-with-web-application-itself/1.1.1.1-provision-user-using-SCIM1.1/src/test/java/org/wso2/identity/scenarios/test/scim/UserProvisionWithInsufficientPrivilegesSCIMTestCase.java]: thanks, used one JSON Object to reduce duplicate codes

**Comment by shavantha** [comment]
closing this PR since refactoring is being done.

**Pull request closed without merging**